chore: replace usage of public class field assignments

boneskull opened the pull request:
Basically, public class fields don't work the way you'd expect, so I'd like to avoid assignment to them whenever possible.  Compounding the situation is the fact that `tsc` (and Babel, fwiw) both transpile in a way that deviates slightly from the official specification.

Given the following example:

```js
class A {
  value = 'a';
  constructor() {
    console.log(this.value);
  }
}

class B extends A {
  value = 'b';
}
```

What is logged by `new B()`?  If you guessed `b`, you'd be both reasonable and wrong.  What's happening is that the super class does not know anything about public class fields in subclasses.  If `value` was instead a method or getter, this _would_ work as you'd expect.

Probably better to avoid them whenever possible.  It is perfectly safe to _declare_ them for type inference & the like, but assignment should be avoided until it's absolutely the right thing to do.

Note that `tsc` has a flag to change the transpilation behavior to act in-line with the spec, but it does not change anything in the above regard, AFAICT.

boneskull commented:
To be clear: `static` fields, private `static` fields, and private class fields do not suffer from this problem

mykola-mokhnach commented on packages/base-driver/lib/basedriver/core.js:
```diff
@@ -56,39 +57,43 @@ class DriverCore {
 
   /** @type {EventHistory} */
-  _eventHistory = {commands: []};
+  _eventHistory;
```
should it be also changed to `#eventHistory` ?

mykola-mokhnach commented on packages/fake-driver/lib/driver.js:
```diff
@@ -15,20 +15,45 @@ const FAKE_DRIVER_CONSTRAINTS = /** @type {const} */ ({
+  appModel;
+
+  /** @type {boolean} */
+  _proxyActive;
```
same as above

mykola-mokhnach commented on packages/universal-xml-plugin/lib/plugin.js:
```diff
@@ -1,20 +1,13 @@
 import log from './logger';
 
 export default class UniversalXMLPlugin extends BasePlugin {
-  commands = [
```
how this assignment should work now?

boneskull replied:
this code was unused afaict.

jlipps replied:
yeah i think it came from an old version of our base plugin logic which required registration of overrides.

boneskull commented:
Current dependencies on/for this PR:
* master
  * **PR #18208** <a href="https://app.graphite.dev/github/pr/appium/appium/18208" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium/18208?utm_source=stack-comment).

jlipps commented on packages/fake-driver/lib/driver.js:
```diff
@@ -205,82 +232,354 @@ export class FakeDriver extends BaseDriver {
-  setAlertText = alert.setAlertText;
-  postAcceptAlert = alert.postAcceptAlert;
-  postDismissAlert = alert.postDismissAlert;
+  /**
```
ugh i hate that one line is now 6. it makes it much less desirable to use this file-splitting pattern

boneskull replied:
I kind of hate it too.  there's another way, which is to just fake it.  let me see if I can do that.

boneskull committed:
feat(types): add orientation type; add optional ELEMENT prop of Element

boneskull committed:
chore(universal-xml-plugin): remove dead code; typecheck all js sources

boneskull committed:
chore(fake-plugin): avoid public class field assignment

boneskull committed:
chore(base-driver): avoid public class field assignment

this should avert potential future bugs when drivers attempt to override these properties

boneskull committed:
chore(appium): avoid public class field assignment

This changes `AppiumDriver` to avoid assignment to class properties, and changes the properties on `Schema` to be actually-private instead of pseudoprivate.

boneskull commented:
I've changed this to exclude changes to `fake-driver`, since they were more involved; see #18237

boneskull commented:
labeled "enhancement" since `@appium/types` gets a new type
